components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java

```diff
@@ -7811,13 +7829,78 @@ public Comment getComment(APIIdentifier identifier, String commentId) throws API
             } catch (SQLException e1) {
                 log.error("Failed to retrieve comment ", e1);
             }
-            handleException("Failed to retrieve comment for API " + identifier.getApiName() + "with comment ID " + commentId, e);
+            handleException("Failed to retrieve comment for API " + apiTypeWrapper.getName() + "with comment ID " +
+                    commentId, e);
         } finally {
             APIMgtDBUtil.closeAllConnections(prepStmt, connection, resultSet);
         }
         return null;
     }
 
+    /**
+     * Returns all the Comments on an API
+     *
+     * @param apiTypeWrapper API type wrapper
+     * @return Comment Array
+     * @throws APIManagementException
+     */
+    public Comment[] getComments(ApiTypeWrapper apiTypeWrapper) throws APIManagementException {
+        List<Comment> commentList = new ArrayList<Comment>();
+        Connection connection = null;
+        ResultSet resultSet = null;
+        PreparedStatement prepStmt = null;
+        Boolean isProduct = apiTypeWrapper.isAPIProduct();
```

Comment:
Use primitive boolean

Reply:
This is removed now. Passing Identifier object to check between API and APIProduct. Fixed in 529ba9909a41f6a98475171e424cb0a0782d19c5